### DISCUSSION TITLE
LED arduino help

### DISCUSSION THREAD

**Discussion by mariamor99** [TheAlbertDev/MASB#44] [Q&A]
Hi Albert! 
I've been trying to see why the code doesn't work properly but I can't find the error. 
When I press the button, the LED starts blinking and if I press it again, it turns off but if I press it once more to start blinking again, it does nothing. It's as if the loop only worked once. 
I also tried putting a _condition = true_ at the end of the code but then it doesn't work at all. 
I attach an image of the code to see if you can guide me a little.
Thank you so much!

```c
bool highToLowTransition = false; // por defecto, su valor sera false --> no apretat
bool condition = true;

void setup() {
  // put your setup code here, to run once:
  pinMode(13, OUTPUT); // configuramos el pin 13 como pin digital de salida
  digitalWrite(13, LOW); // configuramos que al iniciarse el programa el LED este apagado
  pinMode(23, INPUT); // configuramos el pin 23 como pin digital de entrada sin pullup
}

void loop() {
  // put your main code here, to run repeatedly:
  
  if (digitalRead(23) == LOW) { // si el boton es pulsado
    if (highToLowTransition == false) { // y no se habia pulsado antes
      highToLowTransition = true; // guardamos que se ha pulsado
    
    
      while (condition == true) {
        digitalWrite(13, HIGH); // LED encendido
        delay(500); // esperamos medio segundo con el LED encendido
        digitalWrite(13, LOW); // LED apagado
        delay(500); // esperamos medio segundo con el LED apagado
        if (digitalRead(23) == LOW) { // si el boton es pulsado
          if (condition == true) { // y se habia pulsado antes
            condition = false; // guardamos que se ha pulsado
            digitalWrite(13, LOW); // LED apagado
            }                 
       }
     }
    }
  else{
    highToLowTransition = false;


    }
   
  }
  
}
```

**Comment by Albert-Alvarez** [comment]
Hi Maria,

First of all, try to add code in your issues as code. I mean, not using images. In this [page](https://docs.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks) you can see how to do it. Simply, copy/paste your code inside a code block (you can also click in the <kbd>Insert code</kbd> button in the issue editor).

I say this because one thing that I wanted to do was to check if you haver your curly brackets `{` `}` correctly paired, and that's something that I can't to with an image 😅

Besides that, I've seen in your image that you set to false the variable condition (`condition = false`) when you push the button for a second time (to switch off the LED). Doing so, when you push again the button, the condition in the while loop `condition == true` doesn't meet and, then, the LED will not blink. Check this.

Finally, even if you solve this, in the second time that you read the state of the button, you don't check if you've already pushed it previously. You are performing a level-triggered event (not edge-triggered). Check this too.

Don't worry if you are not able to achieve the challenge. I will review it in your Pull Request and provide you more details about how to solve it. The important thing is that you upload your try in other to have a basis on which to base my corrections.

**Comment by mariamor99** [comment]
Ok! I'll try to solve it!
Thank you
